packages/opentelemetry-types/src/context/propagation/Propagation.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { SpanContext } from '../../trace/span_context';
+
+/** Defines a propagation interface. */
+export interface Propagation {
+  /**
+   * Injects the given {@link SpanContext} instance to transmit over the wire.
+   *
+   * OpenTelemetry defines a common set of format values (BinaryFormat and
+   * HTTPTextFormat), and each has an expected `carrier` type.
+   *
+   * @param spanContext the SpanContext to transmit over the wire.
+   * @param format the format of the carrier.
+   * @param carrier the carrier of propagation fields, such as an http request.
+   */
+  inject(spanContext: SpanContext, format: string, carrier: unknown): void;
```

Comment:
What are the backwards compatibility guarantees that opentelem is trying to make? In opentracing-javascript, inject is accepts either a SpanContext or Span. I'm in favor of only allowing a span context since the opentracing code ends up doing `if (context instanceof Span) { context = span.context() }` anyway, but want to make sure the API difference is captured in the event we make a shim down the line.

Reply:
I'm in favor of accepting a `Span` everywhere a `SpanContext` is accepted which makes the API easier to work with.

Reply:
Opened an issue to discuss https://github.com/open-telemetry/opentelemetry-js/issues/58